In [ ]:
import pandas as pd
dfLager = pd.read_excel('Lager.xlsx')



In [1]:
import pandas as pd
df = pd.read_excel('Lager.xlsx')

# Erstelle eine neue Spalte mit der fortlaufenden Nummer
df['H'] = range(1, len(df) + 1)

# Erstelle eine neue Spalte, die angibt, ob sich die 3. Stelle in Column A verändert hat
df['A_changed'] = df['Stellplatz'].apply(lambda x: str(x)[2])

# Erstelle eine neue Spalte, die angibt, ob sich die 3. Stelle in Column A verändert hat
df['H_changed'] = df['H'].diff().fillna(0) != 1

# Erstelle eine neue Spalte, die angibt, ob eine neue Spalte eingefügt werden muss
df['Insert_column'] = df['A_changed'] != df['A_changed'].shift(1) | df['H_changed']

# Füge eine neue Spalte ein, wenn erforderlich
for i, row in df.iterrows():
    if row['Insert_column']:
        df.insert(i, 'Neue Spalte', '')

# Speichere die Daten in einer neuen Excel-Datei

#df.to_excel('deine_neue_datei.xlsx', index=False)

TypeError: unsupported operand type(s) for |: 'str' and 'bool'

In [2]:
import pandas as pd

df = pd.read_excel('Lager.xlsx')

# Erstelle eine neue Spalte mit der fortlaufenden Nummer
df['H'] = range(1, len(df) + 1)

# Erstelle eine neue Spalte, die angibt, ob sich die 3. Stelle in Column A verändert hat
df['A_changed'] = df['Stellplatz'].apply(lambda x: str(x)[2])

# Erstelle eine neue Spalte, die angibt, ob sich die 3. Stelle in Column A verändert hat
df['H_changed'] = df['H'].diff().fillna(0) != 1

# Erstelle eine neue Spalte, die angibt, ob eine neue Spalte eingefügt werden muss
df['Insert_column'] = df['A_changed'] != df['A_changed'].shift(1) | df['H_changed']
df['Insert_column'].fillna(False, inplace=True)  # Fülle NaN-Werte mit False

# Füge eine neue Spalte ein, wenn erforderlich
for i, row in df.iterrows():
    if row['Insert_column']:
        df.insert(i, 'Neue Spalte', '')

# Speichere die Daten in einer neuen Excel-Datei
df.to_excel('Lager_neu.xlsx', index=False)

TypeError: unsupported operand type(s) for |: 'str' and 'bool'

In [ ]:
import pandas as pd
dfLager = pd.read_excel('Lager.xlsx')
dfZUgriffe = pd.read_excel('LT22April2023.XLSX')
dfOrders = pd.read_pickle('prod_Kundenbestellungen.pkl')





In [ ]:
# Filtern Sie die Zugriffe
dfZUgriffe_filtered = dfZUgriffe[dfZUgriffe['Dest.Storage Bin'] == '3202113621']

# Sortieren Sie nach Confirmation Time
dfZUgriffe_filtered = dfZUgriffe_filtered.sort_values(by='Confirmation time.1')

# Fügen Sie die Spalte Laufweg hinzu
dfZUgriffe_filtered['Laufweg'] = range(1, len(dfZUgriffe_filtered) + 1)
#übergebe Laufweg in dfLager
dfMerged = pd.merge(dfLager, dfZUgriffe_filtered, left_on='Stellplatz', right_on='Source Storage Bin', how='left', suffixes=('_lager', '_zugriffe'))

dfLager['Pickzeit'] = dfMerged['Confirmation time.1']# Berechnen Sie die Zugriffe pro Stellplatz
dfLager['Zugriffe'] = dfLager.apply(lambda x: dfZUgriffe_filtered[dfZUgriffe_filtered['Source Storage Bin'] == x['Stellplatz']].shape[0], axis=1)
dfLager['Laufweg'] = dfMerged['Laufweg']# Berechnen Sie die Zugriffe pro Stellplatz

# Berechnen Sie die Zugriffe pro Stellplatz
dfLager['Zugriffe'] = dfLager.apply(lambda x: dfZUgriffe_filtered[dfZUgriffe_filtered['Source Storage Bin'] == x['Stellplatz']].shape[0], axis=1)

In [ ]:
dfLager

Stellplatz Volumen Storagetype Reihe  Platz     X    Y  Unnamed: 7   
0       R01-01    1 EP         SN1   R01      1  2340  830         NaN  \
1       R01-02    1 EP         SN1   R01      2  2240  830         NaN   
2       R01-03    1 EP         SN1   R01      3  2140  830         NaN   
3       R01-04    1 EP         SN1   R01      4  2040  830         NaN   
4       R01-05    1 EP         SN1   R01      5  1940  830         NaN   
..         ...     ...         ...   ...    ...   ...  ...         ...   
156     R07-19    1 EP         SN1   R07     19   540   75         NaN   
157     R07-20    1 EP         SN1   R07     20   440   75         NaN   
158     R07-21    1 EP         SN1   R07     21   340   75         NaN   
159     R07-22    1 EP         SN1   R07     22   240   75         NaN   
160     R07-23    1 EP         SN1   R07     23   140   75         NaN   

     Unnamed: 8 Pickzeit  Zugriffe  Laufweg  
0           NaN      NaN         0      NaN  
1           NaN      NaN         0      NaN  
2           NaN      NaN         0      NaN  
3           NaN      NaN         0      NaN  
4           NaN      NaN         0      NaN  
..          ...      ...       ...      ...  
156         NaN      NaN         0      NaN  
157         NaN      NaN         0      NaN  
158       100.0      NaN         0      NaN  
159         NaN      NaN         0      NaN  
160         NaN      NaN         0      NaN  

[161 rows x 12 columns]

Stellplatz	Volumen	Storagetype	Reihe	Platz	X	Y	Unnamed: 7	Unnamed: 8	Pickzeit	Zugriffe	Laufweg
0	R06-17	2 EP	SN1	R06	17	740	175			10:48:06	0	40
1	R06-18	2 EP	SN1	R06	18	640	175				0	
2	R06-19	2 EP	SN1	R06	19	540	175				0	
3	R06-20	2 EP	SN1	R06	20	440	175				0	
4	R06-21	2 EP	SN1	R06	21	340	175				0	
5	R06-22	2 EP	SN1	R06	22	240	175				0	
6	R06-23	2 EP	SN1	R06	23	140	175				0	
7	R07-01	1 EP	SN1	R07	1	2340	44				0	
8	R07-02	1 EP	SN1	R07	2	2240	44				0	
9	R07-03	1 EP	SN1	R07	3	2140	44				0	
10	R07-04	1 EP	SN1	R07	4	2040	44				1	
11	R07-05	1 EP	SN1	R07	5	1940	44			10:43:49	0	34
12	R07-06	1 EP	SN1	R07	6	1840	44				0	
13	R07-07	1 EP	SN1	R07	7	1740	44				1	
14	R07-08	1 EP	SN1	R07	8	1640	44			09:16:20	0	2
15	R07-09	1 EP	SN1	R07	9	1540	44				1	
16	R07-10	1 EP	SN1	R07	10	1440	44			10:44:08	1	35
17	R07-11	1 EP	SN1	R07	11	1340	44			10:44:35	0	36
18	R07-12	1 EP	SN1	R07	12	1240	44				0	
19	R07-13	1 EP	SN1	R07	13	1140	44				0	
20	R07-14	1 EP	SN1	R07	14	1040	44				0	
21	R07-15	1 EP	SN1	R07	15	940	44				1	
22	R07-16	1 EP	SN1	R07	16	840	44			10:46:30	1	38
23	R07-17	1 EP	SN1	R07	17	740	44			10:47:55	0	39 


Hallo in diesem Datenframe möchte nun vom größten wert bis zum nächst kleineren in Laufweg 1,2 meter pro Row dazwischen in eine neue Spalte Wegstrecke berechnen